config/config.go

```diff
@@ -194,25 +180,26 @@ func WriteCurrentContextToConfigFile(value string) error {
 	return nil
 }
 
-func DecodeYAMLFile(path string) (map[string]string, error) {
-	b, err := ioutil.ReadFile(path)
+func GetHomeDir() string {
+	home, err := homedir.Dir()
 	if err != nil {
-		return nil, fmt.Errorf("could not open %s for parsing. Error: %v", path, err)
+		log.Fatalln("could not get home directory:", err)
 	}
 
-	yf := map[string]string{}
-	err = yaml.Unmarshal(b, yf)
+	return home
+}
+
+// WriteYamlFile writes to the yaml file
+func WriteYamlFile(filename string, value map[string]string) error {
+	marshaled, err := yaml.Marshal(value)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return yf, err
-}
 
-func GetHomeDir() string {
-	home, err := homedir.Dir()
+	err = ioutil.WriteFile(filename, marshaled, ReadWritePerms)
```

Comment:
this would be enough, no need to double-check the error, if it's there, just return that from `ioutil.WriteFile`:
```go
return ioutil.WriteFile(filename, marshaled, ReadWritePerms)
```